src/main/java/org/galatea/starter/service/IexService.java

```diff
@@ -69,6 +79,96 @@ private String setKey() {
     }
   }
 
+  /**
+   * Formats a date from YYYYMMDD format to YYYY-MM-DD format. This is to accommodate the historical
+   * price repository which automatically does this conversion.
+   * @param date A string representing a date in YYYYMMDD format.
+   * @return A string representing the same date in YYYY-MM-DD format.
+   */
+
+  private String formatDate(final String date) {
+    return date.substring(0, 4) + "-" + date.substring(4, 6) + "-" + date.substring(6);
+  }
+
+  /**
+   * Gets historical price data for the given symbol and date from the IEX API. Saves the new
+   * IexHistoricalPrice object to the price repository as a side effect.
+   */
+
+  private List<IexHistoricalPrice> getHistoricalPriceForDate(
+      final String symbol,
+      final String date) {
+
+    String searchDate = formatDate(date);
+    log.info("Searching repository for symbol {} and date {}", symbol, searchDate);
+
+    if (priceRpsy.existsBySymbolAndDate(symbol, searchDate)) {
+      log.info("Historical price for symbol {}, date{} found in repository.", symbol, date);
+      return priceRpsy.findBySymbolAndDate(symbol, searchDate);
+    } else {
+      log.info("Requesting historical price for symbol {}, date {} from IEX.", symbol, date);
+      List<IexHistoricalPrice> prices;
+      prices = iexClient.getHistoricalPriceForDate(symbol, date, true, key);
+      Iterable<IexHistoricalPrice> saved = priceRpsy.saveAll(prices);
+      log.info("Saved historical price to repository: {}", saved);
+      return prices;
+    }
+  }
+
+  private LocalDate rangeToStartDateHelper(final String range, final LocalDate today) {
+
+    ChronoUnit unit = ChronoUnit.DAYS;
+    String rangeSuffix = range.substring(range.length() - 1);
+
+    if (rangeSuffix.equals("m")) {
+      unit = ChronoUnit.MONTHS;
+    } else if (rangeSuffix.equals("y")) {
+      unit = ChronoUnit.YEARS;
+    } else if (!rangeSuffix.equals("d")) {
+      log.warn("Argument range must end with \"d\", \"m\", or \"y\"; received {}.", rangeSuffix);
+      log.warn("Assuming units of DAYS.");
+    }
+
+    int amountToSubtract = Integer.parseInt(range.substring(0, range.length() - 1));
+    return today.minus(amountToSubtract, unit);
```

Comment:
Again, using Date packages can make this much less prone to error...
Take a look at the LocalDate class (hint: ```minusDays```)